### ISSUE TITLE
Narrow Rule::in(...) fluent builder to a literal-string union in validated() output

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#873]
## Problem

A FormRequest rule written with the fluent `Rule::in(...)` builder narrows to `mixed` in `$request->validated()`:

```php
public function rules(): array
{
    return [
        'workday' => ['required', Rule::in('half', 'full')],
    ];
}
```

Psalm then reports a false positive at the call site, because the validated array is inferred as `array{..., workday: mixed}` instead of `array{..., workday: 'half'|'full'}`:

```
ERROR: InvalidArgument
Argument 1 of App\Models\Staff::createDailyUpdate expects array<string, string>,
but array{..., workday: mixed} provided (see https://psalm.dev/004)
    $staff->createDailyUpdate($request->validated());
```

The string form `'in:half,full'` already narrows correctly via `inRuleToLiteralUnion()` (`src/Handlers/Validation/ValidationRuleAnalyzer.php:412`).

## Root cause

`resolveRuleObjectClassName()` (`src/Handlers/Validation/ValidationRuleAnalyzer.php:908`) maps `Rule::in(...)` to `Illuminate\Validation\Rules\In` via `RULE_FACADE_METHOD_RETURN_CLASS` and emits a synthetic `class:Illuminate\Validation\Rules\In` segment.

The synthetic segment is consumed at `ValidationRuleAnalyzer.php:255-261` and, by design, only contributes to the taint-escape bitmask:

```php
// Contributes only to the taint escape bitmask — the rule's runtime behavior
// (type, nullable, required) is opaque to the plugin.
if (\str_starts_with($segment, self::CLASS_SEGMENT_PREFIX)) {
    $removedTaints |= self::classRuleRemovedTaints(...);
    continue;
}
```

When no other type-bearing rule is present, `$type ??= Type::getMixed()` fires (line 298) and the field becomes `mixed`.

Existing test `tests/Type/tests/TaintAnalysis/SafeFormRequestRuleInFluentNoTaint.phpt:19` works around this by spelling `'string'` alongside `Rule::in(...)`.

## Proposed fix

Extract the static-call arguments of `Rule::in(...)` and feed them into the existing `inRuleToLiteralUnion()` helper, so the fluent form narrows the same way the string form does.

Forms to support:

- `Rule::in('a', 'b')` (variadic)
- `Rule::in('a')` (single string)
- `Rule::in(['a', 'b'])` (array literal)

Anything else (enums, variables, arbitrary expressions, `Arrayable` arguments) should fall back to the current `mixed` behavior, no regression.

The same approach can be reused for `Rule::notIn(...)`.

## Out of scope

Generic class→type fallback for argless first-party fluent builders (`Rule::numeric()`, `Rule::email()`, `Rule::date()`, ...) is a follow-up. Those cases are tracked separately if useful.

## Acceptance

- New PHPT type test covering the three argument forms above.
- `validated()` returns `array{..., workday: 'half'|'full'}` for the example above.
- Taint behavior unchanged (`SafeFormRequestRuleInFluentNoTaint.phpt` still passes).
